libs/contrib/Data/Always.idr

```diff
@@ -0,0 +1,39 @@
+module Data.Always
+
+import Data.Elem
+
+%default total
+
+data Always : (x -> Type) -> List x -> Type where
+  Nil : Always f []
+  (::) : f ty -> Always f l -> Always f (ty :: l)
+
+index : Elem x xs -> Always f xs -> f x
+index Here (x :: _) = x
+index (There elem) (_ :: xs) = index elem xs
+
+replaceAt : Elem x xs -> f x -> Always f xs -> Always f xs
+replaceAt Here element (_ :: xs) = element :: xs
+replaceAt (There n) element (x :: xs) = x :: replaceAt n element xs
+
+replaceAndIndex : {elements:Always f xs} -> index elem (replaceAt elem x elements) = x
```

Comment:
There should be spaces around the colon in a type annotation. Also, the compiler is unlikely to solve the `Always f xs` unless you make the argument `auto`.

What have you used this for in the past?